system/Traits/ConditionalTrait.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+/**
+ * This file is part of CodeIgniter 4 framework.
+ *
+ * (c) CodeIgniter Foundation <admin@codeigniter.com>
+ *
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ */
+
+namespace CodeIgniter\Traits;
+
+trait ConditionalTrait
+{
+    /**
+     * Only runs the query when $condition evaluates to true
+     *
+     * @param array|bool|float|int|object|resource|string|null $condition
+     */
+    public function when($condition, callable $callback, ?callable $defaultCallback = null): self
```

Comment:
It seems `static` is better, and please add `@return $this`.

Reply:
I'm confused. `static` function? That wouldn't work in a chain, though, which is the purpose.

Reply:
No, the return type.

Reply:
Sorry, it cannot be used in PHP 7.4.
https://php.watch/versions/8.0/static-return-type